From d2403568808a2e91f245ae18921f9396a10a36bf Mon Sep 17 00:00:00 2001
From: Robert Dobrowolski <robert.dobrowolski@linux.intel.com>
Date: Fri, 13 May 2016 14:18:26 +0200
Subject: [PATCH 1/2] usb: ssic: port id correction in disable stall workaround

Function handle_port_status() is using port_id to determine
the SSIC port which is wrong.
xhci->ssic_port_number is 1 based port number and should be compared
against (faked_port_index + 1), otherwise function implementing
the workaround xhci_change_ssic_regs() is never executed.

Change-Id: I104b724d53ca1e3e8e1b15919527f0e5c31049a9
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-31892
Signed-off-by: Mian Yousaf Kaukab <yousaf.kaukab@intel.com>
Signed-off-by: Robert Dobrowolski <robert.dobrowolski@intel.com>
Reviewed-on: https://android.intel.com:443/505954
---
 drivers/usb/host/xhci-ring.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/usb/host/xhci-ring.c b/drivers/usb/host/xhci-ring.c
index 59e03fb..fd0f21e 100644
--- a/drivers/usb/host/xhci-ring.c
+++ b/drivers/usb/host/xhci-ring.c
@@ -1742,8 +1742,8 @@ static void handle_port_status(struct xhci_hcd *xhci,
 	}
 
 	/* Check for CCS and CSC bits */
-	if (xhci->quirks & XHCI_SSIC_DISABLE_STALL &&
-		port_id == xhci->ssic_port_number) {
+	if ((xhci->quirks & XHCI_SSIC_DISABLE_STALL) &&
+		(faked_port_index + 1) == xhci->ssic_port_number) {
 		/* Check the bit 17 in PORTSC */
 		if (temp & PORT_CSC) {
 			/*
-- 
1.9.1

